src/rate/coinbase.js

```diff
@@ -30,24 +24,24 @@ export function makeCoinbasePlugin(
     },
 
     async fetchRates(pairsHint) {
-      const reply = await io.fetch('https://api.coinbase.com/v2/exchange-rates')
-      const json = await reply.json()
-
-      if (!json || !json.data || !json.data.rates) return []
-
-      // Grab all the USD pairs:
       const pairs = []
-      const keys = Object.keys(json.data.rates)
-      for (const key of keys) {
-        const rate = Number(json.data.rates[key])
-        const toCurrency = fixCurrency(key)
-        pairs.push({
-          fromCurrency: 'iso:USD',
-          toCurrency,
-          rate
-        })
+      for (const pair of pairsHint) {
+        const fiatCode = pair.toCurrency.split(':')
```

Comment:
This will cost a lot of fetches. Better to do the single fetch that returns a lot of pairs but only filter the pairs that are in the hints